Automattic/jetpack
# Social Store | Extract test utility functions and initial data

After #34289, we now have quite a bit of test utilities like `createRegistryWithStores` along with the associated initial data repeated at these places in `publicize-components`:
- [`hooks/use-sync-post-data-to-store/test/index.test.js`](https://github.com/Automattic/jetpack/blob/0b69cd72c0997cc63b532d3760d7ddc914b37f85/projects/js-packages/publicize-components/src/hooks/use-sync-post-data-to-store/test/index.test.js#L76)
- [`social-store/actions/test/connection-data.js`](https://github.com/Automattic/jetpack/blob/33a99c9de5344cf8839cdc8d55bf6007a6e8aa87/projects/js-packages/publicize-components/src/social-store/actions/test/connection-data.js#L75)
- [`utils/test/use-post-meta.test.js`](https://github.com/Automattic/jetpack/blob/33a99c9de5344cf8839cdc8d55bf6007a6e8aa87/projects/js-packages/publicize-components/src/utils/test/use-post-meta.test.js#L56)

This PR removes that duplication and after this PR, adding tests to store logic should be easier.

## Proposed changes:

* Create a reusable `createRegistryWithStores` utility for tests for social store
* Use `createRegistryWithStores` in the existing tests to remove duplication

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* For `publicize-components`, run `jetpack test`
* Confirm that all the tests pass
